gulpfile.js

```diff
@@ -0,0 +1,169 @@
+var taskListing = require('gulp-task-listing');
+var changed = require('gulp-changed');
+var concat = require('gulp-concat');
+
+
+//——————————————————————————————————————————————————————————————————————————————
+// Local debug info
+//——————————————————————————————————————————————————————————————————————————————
+if (process.env.NODE_ENV === 'production') {
+  log(c.bgYellow.black('Production'), c.yellow('environment detected'));
+} else {
+  log(c.bgYellow.black('Local'), c.yellow('environment detected'));
+}
+
+
+//——————————————————————————————————————————————————————————————————————————————
+// BrowserSync
+//——————————————————————————————————————————————————————————————————————————————
+gulp.task('dev:bs', () => {
+  bs({
+    server: './_site',
+    open: false,
+    port: '4000',
+  });
+});
+
+
+//——————————————————————————————————————————————————————————————————————————————
+// Jekyll for development
+//
+// Note: since this site runs on GitHub Pages, the production build of Jekyll is
+// not driven by Gulp. Only GitHub infrastructure can run it.
+//——————————————————————————————————————————————————————————————————————————————
+gulp.task('dev:jekyll', () => {
+  bs.notify('Jekyll building...');
+
+  return spawn('bundle', ['exec', 'jekyll', 'build', '--source=docs', '--config=docs/_config.yml,docs/_config.dev.yml'], {stdio: 'inherit'})
+    .on('close', reload);
+});
+
+
+//——————————————————————————————————————————————————————————————————————————————
+// Sass
+//——————————————————————————————————————————————————————————————————————————————
+gulp.task('dev:sass', ['dev:sass:commondesign', 'dev:sass:ochaextras', 'dev:sass:styleguide']);
+
+gulp.task('dev:sass:commondesign', () => {
+  return sassTask('docs/common-design/css/styles.scss');
+});
+
+gulp.task('dev:sass:ochaextras', () => {
+  return sassTask('docs/ocha/css/extras.scss');
+});
+
+gulp.task('dev:sass:styleguide', () => {
+  return sassTask('docs/styleguide/css/styleguide.scss');
+});
+
+//——————————————————————————————————————————————————————————————————————————————
+// Reusable Sass task. Takes source parameter and outputs to same directory.
+//——————————————————————————————————————————————————————————————————————————————
+function sassTask(source) {
+  bs.notify(`Sass: ${source}`);
+
+  // There are many different Sass directories but we should always compile the
+  // CSS so it's next to the original Sass.
+  let path = source.split('/');
+  let file = path.pop();
+  let dest = path.join('/');
+
+  return gulp.src(source)
+    .pipe(plumber())
+    .pipe(gulpif(process.env.NODE_ENV !== 'production', sourcemaps.init()))
+    .pipe(sass().on('error', sass.logError))
+    .pipe(postcss([
+      prefix({
+        browsers: ['>0.333%', 'iOS 8'],
+        cascade: false,
+      }),
+      // cssnano(),
+    ]))
+    .pipe(gulpif(process.env.NODE_ENV !== 'production', sourcemaps.write('./')))
+    .pipe(gulp.dest(dest))
+    .pipe(gulp.dest('_site/' + dest))
```

Comment:
@left23 the lack of Sass updates til restart probably originates with this line of code, which attempts to dump the compiled CSS into the generated Jekyll site without using Jekyll. I'll look into it.

Reply:
Cool. Changes to the html reloads as expected.